easybuild/easyblocks/q/qscintilla.py

```diff
@@ -0,0 +1,144 @@
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing QScintilla, implemented as an easyblock
+
+author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import apply_regex_substitutions, mkdir, symlink, write_file
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+class EB_QScintilla(ConfigureMake):
+    """Support for building/installing QScintilla."""
+
+    def configure_step(self):
+        """Custom configuration procedure for QScintilla."""
+
+        srcdir = os.path.join(self.cfg['start_dir'], 'Qt4Qt5')
+        try:
+            os.chdir(srcdir)
+        except OSError as err:
+            raise EasyBuildError("Failed to change to %s: %s", srcdir, err)
+
+        regex_subs = [
+            (r'\$\$\[QT_HOST_DATA\]', os.path.join(self.installdir, 'data')),
+            (r'\$\$\[QT_INSTALL_DATA\]', os.path.join(self.installdir, 'data')),
+            (r'\$\$\[QT_INSTALL_HEADERS\]', os.path.join(self.installdir, 'include')),
+            (r'\$\$\[QT_INSTALL_LIBS\]', os.path.join(self.installdir, 'lib')),
+            (r'\$\$\[QT_INSTALL_TRANSLATIONS\]', os.path.join(self.installdir, 'trans')),
+        ]
+        apply_regex_substitutions('qscintilla.pro', regex_subs) 
+
+        run_cmd("qmake qscintilla.pro")
+
+    def build_step(self):
+        """Custom build procedure for QScintilla."""
+
+        # make sure that $CXXFLAGS is being passed down
+        self.cfg.update('buildopts', 'CXXFLAGS="$CXXFLAGS \$(DEFINES)"')
+
+        super(EB_QScintilla, self).build_step()
+
+    def install_step(self):
+        """Custom install procedure for QScintilla."""
+        
+        super(EB_QScintilla, self).install_step()
+
+        # also install Python bindings if Python is included as a dependency
+        python = get_software_root('Python')
+        if python:
+
+            pydir = os.path.join(self.cfg['start_dir'], 'Python')
+            try:
+                os.chdir(pydir)
+            except OSError as err:
+                raise EasyBuildError("Failed to change to %s: %s", pydir, err)
```

Comment:
we should have a `change_dir` function in framework, it's much overdue, but I won't block this PR over it (since I want to include this in the next EasyBuild release which is likely going to be v3.1.2, and `change_dir` wouldn't be supported until v3.2.0)